File: python/QuickSort.py
def partition(arr,l,h): 
	i = ( l-1 )		  
	pivot = arr[h]	 

	for j in range(l , h): 

		if arr[j] < pivot: 
		
			i = i+1
			arr[i],arr[j] = arr[j],arr[i] 

	arr[i+1],arr[h] = arr[h],arr[i+1] 
	return ( i+1 ) 

def sort(arr,l,h): 
	if l < h: 
		pi = partition(arr,l,h) 
		sort(arr, l, pi-1) 
		sort(arr, pi+1, h) 

arr = [12,22,10,2002,1,20,7] 

sort(arr,0,len(arr)-1) 

for x in arr: 
	print (x)

File: c++/QuickSort.cpp
#include <iostream.h> 
using namespace std; 

void swap(int* a, int* b) { 
	int t = *a; 
	*a = *b; 
	*b = t; 
} 
int partition (int arr[], int l, int h) { 
	int pivot = arr[h]; 
	int i = (l - 1); 

	for (int j = l; j <= h - 1; j++) { 
		if (arr[j] < pivot)  { 
			i++; 
			swap(&arr[i], &arr[j]); 
		} 
	} 
	swap(&arr[i + 1], &arr[h]); 
	return (i + 1); 
} 
void sort(int arr[], int l, int h)  { 
	if (l < h)  { 
		int pi = partition(arr, l, h); 
		sort(arr, l, pi - 1); 
		sort(arr, pi + 1, h); 
	} 
} 
int main()  { 
	int arr[] = {12,22,10,2002,1,20,7}; 
	int n = sizeof(arr) / sizeof(arr[0]); 

	sort(arr, 0, n - 1); 

	for (int i = 0; i < size; i++) 
		cout<<arr[i]; 

	return 0; 
} 
